add section on how to get help (#504)
NixOS/nix.dev: source/contributing/how-to-get-help.md

# How to get help

If you need assistance with one of your contributions, there are a few places you
can go for help.
These are sorted from specific to general, and you typically want to start as
specific as possible, since this increases your changes to get help.

Initially, you'll want to try to find individuals or groups with specific knowledge of
your issue:
  - look for the relevant package's maintainers, listed in its `maintainers` attribute

>   - look at the git log for the relevant files

Reviewer: Link to the git manual

  - check out if any teams are responsible for the relevant subsystem:
    - on the [NixOS website](https://nixos.org/community/#governance-teams)
    - in the [team list file](https://github.com/NixOS/nixpkgs/blob/master/maintainers/team-list.nix)

>     - in the [`CODEOWNERS`](https://github.com/NixOS/nixpkgs/blob/master/.github/CODEOWNERS) file

Reviewer: Yes, make each an explicit link

If that fails, you can resort to asking the community at large, using one of the official
communication channels, such as:
  - a related room in the [NixOS Matrix space](https://matrix.to/#/#community:nixos.org)
  - the *Help* category on the [Discourse Forum](https://discourse.nixos.org/c/learn/9)
    - note that people typically (but not always!) have the same username here as on

>       GitHub, so this can be another way of reaching out to specific maintainers.

Reviewer 2: This should be in the above more specific section, the different ways in which you can contact specific people, Email, Discourse and Matrix should be mentioned, ideally the ones from the maintainers.nix file

# How to get help

If you need assistance with one of your contributions, there are a few places you
can go for help.

For better efficiency and chances of success, you should try to contact individuals or groups with more specific knowledge first:
- Look at the [git log](https://www.git-scm.com/docs/git-log) for the files you need help
  with, and take note of the email addresses of people who committed.
- Check out if any teams are responsible for the relevant subsystem:
  - On the [NixOS website](https://nixos.org/community/#governance-teams).

>   - In the [team list
>     file](https://github.com/NixOS/nixpkgs/blob/master/maintainers/team-list.nix).

Reviewer: ```suggestion
  - In the [Nixpkgs maintainer teams list](https://github.com/NixOS/nixpkgs/blob/master/maintainers/team-list.nix).
```

>   - In the [Nixpkgs](https://github.com/NixOS/nixpkgs/blob/master/.github/CODEOWNERS) or
>     [Nix](https://github.com/NixOS/nix/blob/master/.github/CODEOWNERS) `CODEOWNERS` files.

Reviewer: ```suggestion
  - In the [Nixpkgs `CODEOWNERS`](https://github.com/NixOS/nixpkgs/blob/master/.github/CODEOWNERS) or
    [Nix `CODEOWNERS`](https://github.com/NixOS/nix/blob/master/.github/CODEOWNERS) files.
```

> - If your contribution is for a nixpkgs package, look for its maintainers in the
>   [`maintainers`](https://nixos.org/manual/nixpkgs/stable/#var-meta-maintainers)
>   attribute.

Reviewer: ```suggestion
- If your contribution is for a Nixpkgs package, look for its maintainers in the [`maintainers`](https://nixos.org/manual/nixpkgs/stable/#var-meta-maintainers) attribute.
```

Author: Is this also about indentation of the second line? Because my editor (Emacs) wants to do it like this, not sure if I should go and tweak it.

> Once you have found the people you're looking for (either their email address, or their
> handles in [`maintainers-list.nix`][maintainers-list]), you can contact them via:

Reviewer: ```suggestion
Once you have found the people you're looking for, you can contact them via:
```

Since we go into details immediately, no need to repeat that specific information

> - Email.
> - [Matrix], using the Matrix handle found in the [`maintainers-list.nix`
>   file][maintainers-list]. If no Matrix handle is present for a specific maintainer, try
>   searching for them using their GitHub username, as most people tend to use the same one
>   across channels.
> - [Discourse](https://discourse.nixos.org), again trying the GitHub username found in the
>   [`maintainers-list.nix` file][maintainers-list] to search for a specific user.

Reviewer: Probably email should be the last resort.

```suggestion
- GitHub

  [Mention the GitHub username](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#mentioning-people-and-teams) found in the [`maintainers-list.nix` file][maintainers-list] in issue or pull request descriptions, or comments.

- [Discourse](https://discourse.nixos.org)

  Try the GitHub username found in the [`maintainers-list.nix` file][maintainers-list] mention or directly contact a specific user.
  Note that some people use a different username on Discourse.
  
- [Matrix]

  Use the Matrix handle found in the [`maintainers-list.nix` file][maintainers-list].
  If no Matrix handle is present for a specific maintainer, try searching for them using their GitHub username, as most people tend to use the same one across channels.
  
- Email

  Use email addresses found with `git log` as a last resort.
```

If you haven't found any specific users or groups that could help you with your
contribution, you can resort to asking the community at large, using one of the following official
communication channels:

- A room related to your question in the [NixOS Matrix space][matrix].

> - The *Help* category on the [Discourse Forum](https://discourse.nixos.org/c/learn/9).

Reviewer: ```suggestion
- The [Help* category on Discourse](https://discourse.nixos.org/c/learn/9).
```
